tests/lib/rules/no-restricted-globals.js

```diff
@@ -60,122 +60,210 @@ ruleTester.run("no-restricted-globals", rule, {
-            errors: [{ message: "Unexpected use of 'foo'.", type: "Identifier" }]
+            errors: [{
+                messageId: "defaultMessage",
+                data: { name: "foo" },
+                type: "Identifier"
+            }]
         },
         {
             code: "foo",
             options: [{ name: "foo" }],
-            errors: [{ message: "Unexpected use of 'foo'.", type: "Identifier" }]
+            errors: [{
+                messageId: "defaultMessage",
+                data: { name: "foo" },
+                type: "Identifier"
+            }]
         },
         {
             code: "function fn() { foo; }",
             options: [{ name: "foo" }],
-            errors: [{ message: "Unexpected use of 'foo'.", type: "Identifier" }]
+            errors: [{
+                messageId: "defaultMessage",
+                data: { name: "foo" },
+                type: "Identifier"
+            }]
         },
         {
             code: "function fn() { foo; }",
             options: [{ name: "foo" }],
             globals: { foo: false },
-            errors: [{ message: "Unexpected use of 'foo'.", type: "Identifier" }]
+            errors: [{
+                messageId: "defaultMessage",
+                data: { name: "foo" },
+                type: "Identifier"
+            }]
         },
         {
             code: "event",
             options: ["foo", { name: "event" }],
             env: { browser: true },
-            errors: [{ message: "Unexpected use of 'event'.", type: "Identifier" }]
+            errors: [{
+                messageId: "defaultMessage",
+                data: { name: "event" },
+                type: "Identifier"
+            }]
         },
         {
             code: "foo",
             options: [{ name: "foo" }],
             globals: { foo: false },
-            errors: [{ message: "Unexpected use of 'foo'.", type: "Identifier" }]
+            errors: [{
+                messageId: "defaultMessage",
+                data: { name: "foo" },
+                type: "Identifier"
+            }]
         },
         {
             code: "foo()",
             options: [{ name: "foo" }],
-            errors: [{ message: "Unexpected use of 'foo'.", type: "Identifier" }]
+            errors: [{
+                messageId: "defaultMessage",
+                data: { name: "foo" },
+                type: "Identifier"
+            }]
         },
         {
             code: "foo.bar()",
             options: [{ name: "foo" }],
-            errors: [{ message: "Unexpected use of 'foo'.", type: "Identifier" }]
+            errors: [{
+                messageId: "defaultMessage",
+                data: { name: "foo" },
+                type: "Identifier"
+            }]
         },
         {
             code: "foo",
             options: [{ name: "foo", message: "Use bar instead." }],
-            errors: [{ message: "Unexpected use of 'foo'. Use bar instead.", type: "Identifier" }]
```

Comment:
Could be nice to assign `"Use bar instead"` to a variable we can reuse in this file.